wp/lib/bap_wp/src/z3_utils.ml

```diff
@@ -100,6 +100,113 @@ let get_z3_name (map : Constr.z3_expr Var.Map.t) (name : string) (fmt : Var.t ->
         else
           None)
 
+(** Sexp.t helper constructors for smtlib constructs *)
+let define_fun name args retsort body =
+  let args = List.map ~f:(fun (v,sort) -> Sexp.List [v; sort]) args in
+  Sexp.List [Sexp.Atom "define-fun"; Sexp.Atom name; Sexp.List args; retsort; body]
+let synonym name ret body = define_fun name [] ret body
+let define_sort (name : string) s = Sexp.List [
+ Sexp.Atom "define-sort"; Sexp.Atom name; Sexp.List []; s]
+let sexp_of_sort (s : Z3.Sort.sort) : Sexp.t = Sexp.of_string (Z3.Sort.to_string s)
+
+(* [make_arg_synonyms] constructs smtlib formula for the arguments of a subroutine.
+   This make synonyms that for high level c-like names ot low level locations *)
```

Comment:
Is `ot` supposed to be `at`?

Reply:
I have to admit I still don't really get this sentence.

Maybe an example here?